src/test/java/com/datadog/api/v2/client/api/SecurityMonitoringApiTest.java

```diff
@@ -192,121 +194,181 @@ public void searchSignals() throws Exception {
+                OffsetDateTime secondTimestamp = responseDescending.getData().get(1).getAttributes().getTimestamp();
+                return firstTimestamp.isAfter(secondTimestamp);
+            } catch (ApiException ignored) {
+                return false;
+            }
+        });
 
         // Paging
-        SecurityMonitoringSignalsListResponse pageOneResponse = api.searchSecurityMonitoringSignals()
-                .body(new SecurityMonitoringSignalListRequest()
-                        .filter(allSignalsFilter)
-                        .page(new SecurityMonitoringSignalListRequestPage().limit(1)))
-                .execute();
-        assertEquals(1, pageOneResponse.getData().size());
+        TestUtils.retry(10, 10, () -> {
+            try {
+                SecurityMonitoringSignalsListResponse pageOneResponse = api.searchSecurityMonitoringSignals()
+                        .body(new SecurityMonitoringSignalListRequest()
+                                .filter(allSignalsFilter)
+                                .page(new SecurityMonitoringSignalListRequestPage().limit(1)))
+                        .execute();
 
-        String cursor = pageOneResponse.getMeta().getPage().getAfter();
-        assertTrue(pageOneResponse.getLinks().getNext().contains(URLEncoder.encode(cursor)));
+                if (pageOneResponse.getData() == null || pageOneResponse.getData().size() < 1)
+                    return false;
 
-        SecurityMonitoringSignalsListResponse pageTwoResponse = api.searchSecurityMonitoringSignals()
-                .body(new SecurityMonitoringSignalListRequest()
-                        .filter(allSignalsFilter)
-                        .page(new SecurityMonitoringSignalListRequestPage()
-                                .cursor(cursor)
-                                .limit(1)))
-                .execute();
-        assertEquals(1, pageTwoResponse.getData().size());
+                String cursor = pageOneResponse.getMeta().getPage().getAfter();
+                boolean correctFirstLink = pageOneResponse.getLinks().getNext().contains(URLEncoder.encode(cursor));
+
+                SecurityMonitoringSignalsListResponse pageTwoResponse = api.searchSecurityMonitoringSignals()
+                        .body(new SecurityMonitoringSignalListRequest()
+                                .filter(allSignalsFilter)
+                                .page(new SecurityMonitoringSignalListRequestPage()
+                                        .cursor(cursor)
+                                        .limit(1)))
+                        .execute();
+                if (pageTwoResponse.getData() == null || pageTwoResponse.getData().size() < 1)
+                    return false;
+                
+                boolean correctSecondLink = !pageOneResponse.getData().get(0).getId().equals( pageTwoResponse.getData().get(0).getId());
```

Comment:
We are testing first link of first page is not the same as first link on second page. `correctSecondLink` is a confusing name IMO

Reply:
indeed, I corrected it!